### PR TITLE
Re-wrote chi-square, removed dropping buckets.

### PR DESCRIPTION
**Pull request by daryadedik**
As discussed I removed category dropping (now we don't drop any categories), min_counts is input parameter (I set it to 5 as default), but in ADS afterward we will decide which threshold to use.
Also added number of filtered entities per variant in exp.metadata.

### CODE REVIEW

**Comment by shansfolder** [comment]
@daryadedik I think I need a meeting in person to understand this PR ;)

File: expan/core/experiment.py [hunk 1]
```diff
@@ -319,11 +319,12 @@ def chi_square_test_result_and_statistics(self, variant_column, weights, min_cou
...
-        variants_counts = variant_column.value_counts()
+        observed_freqs = variant_column.value_counts()
 
-        # Ensure at least a frequency of 5 at every location in observed_counts, otherwise drop the category
-        # see http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
-        observed_freqs = variants_counts[variants_counts >= min_counts]
+        # Ensure at least a frequency of 5 at every location in observed_counts.
+        # It's recommended to not conduct test if frequencies in each category is less than 5
+        if len(observed_freqs[observed_freqs < min_counts]) >= 1:
+            raise ValueError("Chi-square test is not valid for small expected or observed frequencies.")
```
Comment (igusher) [inline expan/core/experiment.py]: @daryadedik @gbordyugov So what is a canonical way in Python to test input parameters?
asserts? or if-raise?

Comment (igusher) [reply expan/core/experiment.py]: as I see it for a second time already, I tag you both

**Comment by daryadedik** [comment]
Ok, as discussed I just returned previous chi-square implementation (nothing changed here), plus some small code beatifying made, and removed category dropping (now we don't drop any categories), min_counts is input parameter (I set it to 5 as default), but in ADS afterward we will decide which threshold to use.

**Comment by daryadedik** [comment]
Sorry for mixing up the PR. I also added number of filtered entities per variant in exp.metadata. Will need that information instead computing the filtered number on ADS side.

**Review by shansfolder — APPROVED** [review]

**Pull request merged**